torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -514,22 +514,25 @@ bool ProcessGroupAgent::handleRecv(RecvWork& work) {
       std::move(data.first), std::move(data.second), work.type_, work.id_);
   if (message.isRequest()) {
     ++serverActiveCalls_;
-    std::shared_ptr<FutureMessage> futureResponse;
+    auto futureResponse =
+        std::make_shared<JitFuture>(at::AnyClassType::get());
```

Comment:
These lines are different than the ones they're replacing: before we were creating an empty (null) shared_ptr, here we create a "full" shared_ptr that points to an uncompleted future. It also appears that immediately after we re-set this shared_ptr to a new value, hence destroying that uncompleted future. I think it makes sense to avoid creating it in the first place.